doc/api.md

```diff
@@ -0,0 +1,120 @@
+# API
+
+## cartodb.DI.createDashboard(domObject, vizJson, callback(err, obj))
+Creates a new dashboard inside `domObject` based on `vizJson`.
+
+#### Arguments
+- domObject: dom object identifier where the dashboard will be inserted
+- vizJson: either a vizjson url or a vizjson object. See viz.json spec to see how to generate one
+- callback: it will be called when the dashboard is generated. First argument will be null if
+  everything was fine, an `Error` object otherwise. Second argument is a `Dashboard` object, see
+  below
+
+#### Example
+```js
+cartodb.deepInsights.createDashboard('#dashboard', vizJSONurl, function(err, dashboard) {
+  if (err) {
+    console.log('there was an error generating the dashboard');
+    return;
+  }
+  var map = dashboard.getMap()
+});
+```
+
+## cartodb.DI.Dashborbard
+
+This object contains a CartoDB dashboard, a map, and some widgets:
+
+##### getWidgets() -> Array of widgets
+##### getWidget(id) -> get widget
+##### getMap() -> returns a Map object
+
+##### createCategoryWidget(widgetAttrs) -> CategoryWidget
+
+*example*
+```js
+var map = dashboard.getMap();
+var params = {
+  "type": "category",
+  "title": "Metro line",
+  "type": "aggregation",
+  "column": "closest_metro_line",
+  "aggregationColumn": "closest_metro_line",
+  "aggregation": "count",
+};
+// adds a category widget using column `test` from the second layer
+dashboard.createCategory(params, map.getLayer(2))
+```
+
+
+##### createHistogramWidget(widgetAttrs, layer) -> HistogramWidget
+`widgetAttrs` is an object with the following attributes: 
+
+Mandatory attributes:
+- `title`: title shown in the widget
+- `bins`: number of histogram bins
+- `column`: column to generate the histogram
+- `operation`: `count` by default, it could be `avg`
+
+Optional:
+- `id`: identifier that will be used when calling `getWidget`
+- `order`: order inside the widget list
+- `collapsed`: show the wiget collapsed if true
+- `show_stats`: show histogram stats
+- `normalized`: normalize data seen in the map bounding box with the global dataset histogram 
+
+##### createFormulaWidget(widgetAttrs, layer) -> FormulaWidget
+widgetAttrs: `id`, `title`, `order`, `collapsed`, `prefix`, `suffix`, `show_stats`, `description`
```

Comment:
It seems that they are documented thoroughly for createHistogramWidget but not so well document for the rest of these methods. You can decide what level of detail is best at this point, but It'd be nice to keep it consistent.